torch/ao/quantization/quantizer/qnnpack_quantizer.py

```diff
@@ -0,0 +1,145 @@
+from torch.ao.quantization.backend._qnnpack_pt2e import get_qnnpack_pt2e_backend_config
+
+from .quantizer import Quantizer
+
+import typing
+import copy
+from dataclasses import asdict, dataclass
+from typing import Callable, Tuple, Union
+
+import torch
+
+SUPPORTED_DTYPES = [torch.uint8, torch.int8, torch.int32]
+SUPPORTED_QSCHEMES = [
+    torch.per_tensor_affine,
+    torch.per_channel_affine,
+    torch.per_channel_symmetric,
+    torch.per_channel_affine_float_qparams,
+]
+
+
+@dataclass
+class TensorSpec:
```

Comment:
Why is this called tensorSpec. It should be QuantizationSpec.

Reply:
this is initially called QuantizationParam, and OperatorSpec is called QuantizationSpec, feels a bit confusing though, but this file is supposed to be a starting point/prototype, feel free to modify since right now qnnpack/xnnpack is fully owning this

Reply:
can you postpone moving to base quantizer a bit, I would like to discuss with other people and gather more user data from backend developers before deciding

Reply:
having annotate and validate as the only thing in base quantizer makes no sense to me. We at least have to enforce implementation of quantizer to expose, what quantization spec and modules/op they support.

Reply:
>  would like to discuss with other people and gather more user data from backend developers before deciding

FOr this can you concretely identify what the concerns are?

Reply:
clarified in chat, I'm fine with either as long as we are open to change according to data in the future